lago/subnet_lease.py

```diff
@@ -17,222 +17,505 @@
+            LOGGER.debug('Removed subnet lease {}'.format(lease.path))
+
+    def _lease_owned(self, lease, current_uuid_path):
+        """
+        Checks if the given lease is owned by the prefix whose uuid is in
+        the given path
+
+        Note:
+            The prefix must be also in the same path it was when it took the
+            lease
+
+        Args:
+            path (str): Path to the lease
+            current_uuid_path (str): Path to the uuid to check ownership of
+
+        Returns:
+            bool: ``True`` if the given lease in owned by the prefix,
+                ``False`` otherwise
+        """
+
+        prev_uuid_path, prev_uuid = lease.metadata
+
+        with open(current_uuid_path) as f:
+            current_uuid = f.read()
+
+        return \
+            current_uuid_path == prev_uuid_path and \
+            prev_uuid == current_uuid
+
+    def create_lease_object_from_idx(self, idx):
+        """
+        Create a lease from self._subnet_template and put idx as its third
+        octet.
+
+        Args:
+            idx (str): The value of the third octet
+
+        Returns:
+            Lease: Lease object which represents the requested subnet.
+
+        Raises:
+            LagoSubnetLeaseOutOfRangeException: If the resultant subnet is
+            malformed or out of the range of the store.
+        """
+
+        return self.create_lease_object_from_subnet(
+            self._subnet_template.format(idx)
+        )
+
+    def create_lease_object_from_subnet(self, subnet):
+        """
+        Create a lease from ip in a dotted decimal format,
+        (for example '192.168.200.0/24'). the _cidr will be added if not exist
+        in 'subnet'.
+
+        Args:
+            subnet (str): The value of the third octet
+
+        Returns:
+            Lease: Lease object which represents the requested subnet.
+
+        Raises:
+            LagoSubnetLeaseOutOfRangeException: If the resultant subnet is
+            malformed or out of the range of the store.
+        """
+        if '/' not in subnet:
+            subnet = '{}/{}'.format(subnet, self._cidr)
+        if not self.is_leasable_subnet(subnet):
+            raise LagoSubnetLeaseOutOfRangeException(
+                subnet, self.get_allowed_range()
+            )
+
+        return Lease(store_path=self.path, subnet=subnet)
+
+    def is_leasable_subnet(self, subnet):
+        """
+        Checks if a given subnet is inside the defined provision-able range
+
+        Args:
+            subnet (str): Ip in dotted decimal format with _cidr notation
+                (for example '192.168.200.0/24')
+
+        Returns:
+            bool: True if subnet can be parsed into IPNetwork object and is
+                inside the range, False otherwise
+        """
+        try:
+            return \
+                self._min_subnet <= \
+                IPNetwork(subnet) <= \
+                self._max_subnet
+        except AddrFormatError:
```

Comment:
Fixed. 
Also separated the test to two, one for malformed address and the other for out of range address.